### ISSUE TITLE
Is it possible to add CTR/CustomCSSforFx into a potential Cyberfox 60+ release?

### ISSUE THREAD

**Issue by Peacock365** [Aris-t2/CustomCSSforFx#116]
Hi there!

As the title already says, this thread is about potential ways to get CTR working in a newer Cyberfox release. The Cyberfox creator is considering to create a build out of the next Firefox ESR release (version 60), according to this:

> "Have been considering depending on how things are for me at the time, maybe can not make any promises what so ever, doing the next ESR release however it won't be the same as current due to changes made."

quoted from here: https://8pecxstudios.com/Forums/viewtopic.php?f=6&t=1756&start=70#p12212

He has recently reaffirmed this as a possibility, but I couldn't find the more recent quote in their support forums.

I proposed three potential ways for an implementation of CTR in a current Gecko-based browser:

- Using this repo, trying to get the code built-in into Firefox 60.
- Adding CTR as an entry to the _extensions.legacy.exceptions_ setting. This would for sure break things; the breakage needs to be investigated, and CTR surely would have to be modified.
- Embedding CTR in the Basilisk browser, and use this as a base. 

Since CyberCTR (forked from CTR) is the most distinctive feature of Cyberfox, losing it would be a shame.

@Aris-t2, what are the possibilities to include either this repo or the add-on into Cyberfox 60?

I cross-opened an issue here: https://github.com/InternalError503/Language-Manager/issues/17

**Comment by Aris-t2** [comment]
It does not sound to me like Cyberfox dev would like to add this himself, right?

_1. Using this repo, trying to get the code built-in into Firefox 60._
This is definitively possible. Everything called from userChrome.css could just be added to the internal browser.css for example. One problem would remain though: the ability to customize all available features. Meaning someone would have to decide what gets hardcoded into the project. Setting where you want your tabs or whether you like default tabs or classic squared tab would not be possible, if Cyberfox does not add own settings for stuff like that. 

_2. Adding CTR as an entry to the extensions.legacy.exceptions setting. This would for sure break things; the breakage needs to be investigated, and CTR surely would have to be modified._
CTR did not work on Firefox 57+ Nightly, because overlay add-ons support was dropped alongside with xul/xpcom support. I doubt adding it to exceptions would fix this issue. Furthermore all the code would require updating the way it was updated for this project.

_3. Embedding CTR in the Basilisk browser, and use this as a base._
This should work.

**Comment by mazesy** [comment]
Hi everyone!

Maybe not a request because the suggestion isn't really vital.

I have in mind `searchbar_popup_engines_show_labels.css` and the idea of associating with the search engines, after their label, their keyword (if/when applicable of course).

The following pic shows how the [Omnibar Lite](https://addons.mozilla.org/en-US/firefox/addon/omnibar-lite/) extension added to the available engines their keywords.
_Omnibar Lite_ is/was a legacy add-on but I refer to it in order to illustrate my suggestion:

![omnibar lite](https://user-images.githubusercontent.com/31442711/39082742-3280ca0a-4558-11e8-9fd8-3f5aca0a3c05.png)

Now, with searchbar_popup_engines_show_labels.css I get:

![searchbar](https://user-images.githubusercontent.com/31442711/39082749-627dc4d8-4558-11e8-9d70-bf345731748e.jpg)

Having the keyword associates, after the engines' names would be helpful as a fast search engines' keywords reminder. I'm not sure this is feasible with css, maybe requires javascript. You get the idea!

Thanks.

**Comment by Aris-t2** [comment]
It is possible to modify the shown label for it to show a "keyword" after engine name. It is only an illustration and has no actual functionality. Keywords will have to be created for every search engine manually. 


`content: attr(tooltiptext) ' (g)' !important;` <-> `content: '(g) ' attr(tooltiptext) !important;`
(Keywords can only be placed directly before or after the label, not at lines end or similar)

Demo:
![1](https://user-images.githubusercontent.com/6013569/39097060-4c5c4116-4657-11e8-80e4-a14078d6d1b6.png)

Code:
```css
#PopupSearchAutoComplete .search-panel-one-offs .searchbar-engine-one-off-item[tooltiptext*="Google"]::after {
  content: attr(tooltiptext) ' (g)' !important;
}

#PopupSearchAutoComplete .search-panel-one-offs .searchbar-engine-one-off-item[tooltiptext*="Bing"]::after {
  content: attr(tooltiptext) ' (b)' !important;
}

#PopupSearchAutoComplete .search-panel-one-offs .searchbar-engine-one-off-item[tooltiptext*="DuckDuckGo"]::after {
  content: attr(tooltiptext) ' (dd)' !important;
}

#PopupSearchAutoComplete .search-panel-one-offs .searchbar-engine-one-off-item[tooltiptext*="Wikipedia"]::after {
  content: attr(tooltiptext) ' (wi)' !important;
}

#PopupSearchAutoComplete .search-panel-one-offs .searchbar-engine-one-off-item[tooltiptext*="Amazon"]::after {
  content: attr(tooltiptext) ' (a)' !important;
}

#PopupSearchAutoComplete .search-panel-one-offs .searchbar-engine-one-off-item[tooltiptext*="Twitter"]::after {
  content: attr(tooltiptext) ' (t)' !important;
}
```

**Comment by mazesy** [comment]
Thanks @Aris-t2 ... better than nothing but tedious with many search engines. I was hoping the keyword could be extracted together with the search engine's label.

The work-around you suggest definitely works, no problem, but as you write it, "_Keywords will have to be created for every search engine manually_.".  Oh well, no issue when cosmetic/hardly useful.

Thanks.

**Comment by Aris-t2** [comment]
Sadly search engines to not offer the "alias" tag anymore in Fx 57+ (like Omnibar Lite used). The tag is always empty, otherwise it would be possible to add keywords to "old search" at least.

![1](https://user-images.githubusercontent.com/6013569/39126247-20978a4e-4702-11e8-8a1a-507195220960.png)

**Comment by mazesy** [comment]
> Sadly search engines to not offer the "alias" tag anymore in Fx 57+ (like Omnibar Lite used). The tag is always empty, otherwise it would be possible to add keywords to "old search" at least.

That explains it. It's always nicer to understand the reasons. Thanks for detailing them, @Aris-t2 
I have 30+ search engines, I guess I'll spare a few valuable minutes to insert them manually nevertheless.

As I mentioned above, I do refer to search engines with your `searchbar_popup_engines_show_labels.css` because I chose to have the Searchbar on the GUI. A recent choice considering I've opted for the _Awesomebar_ built-in search engines ever since made available, but that I finally prefer the advantages of a visible Searchbar and moreover with search engines labels clearly visible, as `searchbar_popup_engines_show_labels.css` makes it.
This doesn't stop me from searching directly from the Locationbar with search engine keyword followed by my query. But I happen sometimes to forget the keywords of the less used search engines :=)

Thanks again!
